firedrake/checkpointing.py

```diff
@@ -288,3 +289,161 @@ def __del__(self):
         if hasattr(self, "comm"):
             free_comm(self.comm)
             del self.comm
+
+
+class HDF5File(object):
+
+    """An object to mimic the DOLFIN HDF5File class.
```

Comment:
The role of the object is to facilitate checkpointing or similar. The fact that it's DOLFIN compatible is secondary for most users.